Sources/MuxPlayerSwift/FairPlay/ContentKeySessionDelegate.swift

```diff
@@ -5,125 +5,155 @@
-            print("!! Error: URI [\(uri)] did not have playbackId!")
+            logger.debug("\(#function) Error: cannot parse key URL [\(uri)]")
             return nil
         }
-        print("|| PlaybackID from \(uri) is \(playbackID)")
-        return playbackID
+
+        return urlComponents.findQueryValue(
+            key: "playbackId"
+        )
     }
     
     func handleContentKeyRequest(request: any KeyRequest) {
-        print("<><>handleContentKeyRequest: Called")
+        logger.debug(
+            "Called \(#function)"
+        )
+
+        guard let sessionManager = self.sessionManager else {
+            // TODO: Should this also invoke `processContentKeyResponseError`?
+            logger.debug("Missing session manager")
+            return
+        }
+
         // for hls, "the identifier must be an NSURL that matches a key URI in the Media Playlist." from the docs
-        guard let keyURLStr = request.identifier as? String,
-              let keyURL = URL(string: keyURLStr),
-              let assetIDData = keyURLStr.data(using: .utf8)
+        guard let requestIdentifierString = request.identifier as? String,
+              let mediaPlaylistKeyURL = URL(string: requestIdentifierString),
+              let utfEncodedRequestIdentifierString = requestIdentifierString.data(using: .utf8)
         else {
-            print("request identifier was not a key url, this is exceptional for hls")
+            // TODO: Should this also invoke `processContentKeyResponseError`?
+            logger.debug(
+                "CK request identifier not a valid key url."
+            )
             return
         }
-        
-        let playbackID = parsePlaybackId(fromSkdLocation: keyURL)
-        guard let playbackID = playbackID else {
+
+        guard let playbackID = parsePlaybackId(
+            fromSkdLocation: mediaPlaylistKeyURL
+        ) else {
             request.processContentKeyResponseError(
                 FairPlaySessionError.unexpected(
                     message: "playbackID not present in key uri"
                 )
             )
-            return
-        }
-        
-        guard let sessionManager = self.sessionManager else {
-            print("no session manager")
+            logger.debug("\(#function) Error: key url SDK location missing playbackId [\(mediaPlaylistKeyURL.absoluteString)]")
             return
         }
 
-        let playbackOptions = sessionManager.findRegisteredPlaybackOptions(
+        guard let playbackOptions = sessionManager.findRegisteredPlaybackOptions(
             for: playbackID
-        )
-        guard let playbackOptions = playbackOptions,
-              case .drm(let drmOptions) = playbackOptions.playbackPolicy else {
-            print("DRM Tokens must be registered when the AVPlayerItem is created, using FairplaySessionManager")
+        ), case .drm(let drmOptions) = playbackOptions.playbackPolicy else {
+            logger.debug(
```

Comment:
nit: is this a case for `.fault` instead of debug?

Reply:
That is a good question. Went with [`debug`](https://developer.apple.com/documentation/os/oslogtype/2320717-debug) level across the board because it's intended for capturing issues during development and is not persisted to disk.

![Screenshot 2024-06-27 at 9 53 05 AM](https://github.com/muxinc/mux-player-swift/assets/102617203/770bd3d0-1d6c-4f5f-81d6-35ca7295ed4f)

https://developer.apple.com/documentation/os/logging/generating_log_messages_from_your_code#3665947

Other levels are heavier, e.g. `fault` will persist to disk not just the log itself but also additional metadata. That's not to say we shouldn't use them where we think they're needed.

Reply:
Understood!